### PR TITLE
Adds tests

### LINKED ISSUE #15: Multiple test files that both rely on global variables
At the moment we have two test files that both rely on a few global variables (document, window, navigator). When the tests are run as a set they fail because the two tests interfere with each other.

So far we've remedied this by defining the globals in an index.js file, then calling the two tests from the index. A much better solution is to find a way to reset the globals between tests.

Will update when we've had more of a play around.

Update: Adding a global reset to each file and then setting the file-specific globals __inside__ the relevant tests fixed the issue because tests are always synchronous.

### PR DESCRIPTION
**Pull request by finnhodgkin**
Adds tests and brings coverage up to 100%.

Not absolutely happy with the methods we use to achieve this coverage, see #18 and #14

Fix #15 #13 #12 #9

### CODE REVIEW

File: example-complete/public/script.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ function dec(model) {
 // Update both the counter and the model
 function update(newModel, element) {
   model = newModel;
-  element.innerText = model;
+  element.textContent = model;
```
Comment (nelsonic) [inline example-complete/public/script.js:51]: Is the lack of `innerText` support in JSDOM documented somewhere?

File: example-complete/public/script.js [hunk 2]
```diff
@@ -66,17 +66,26 @@ decDom.addEventListener('click', function() {
 var onlineStatusDom = document.querySelector('.online-status');
 // navigator.onLine will be true when online and false when offline. We update the text in the online status element in the dom to reflect the online status from navigator.onLine
 if (navigator.onLine) {
-  onlineStatusDom.innerText = 'online';
+  onlineStatusDom.textContent = 'online';
```
Comment (nelsonic) [inline example-complete/public/script.js:69]: this _could_ be a `function` to reduce code duplication below.

File: example/public/script.js [hunk 3]
```diff
@@ -4,6 +4,17 @@
 var incDom = document.querySelector('.increment');
 var decDom = document.querySelector('.decrement');
 var count = document.querySelector('.count');
+
+// Increment the counter when there's a click event on the + button
+incDom.addEventListener('click', function() {
```
Comment (nelsonic) [inline example/public/script.js:9]: no harm in borrowing the code from: https://github.com/dwyl/learn-elm-architecture-in-javascript for this.  so the tutorials have continuity. 😉

File: example/public/script.js [hunk 4]
```diff
@@ -20,15 +31,14 @@ function dec(model) {
...
-// Increment the counter when there's a click event on the + button
-incDom.addEventListener('click', function() {
-  update(inc(model), count);
-});
-
-// decrement the counter when there's a click event on the - button
-decDom.addEventListener('click', function() {
-  update(dec(model), count);
-});
+/* istanbul ignore next */
```
Comment (nelsonic) [inline example/public/script.js:37]: as discussed. 👍

File: test/exampleComplete.spec.js [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+
+  const updateReadyEvent = new window.applicationCache.Event('updateready');
+
+  window.applicationCache.dispatchEvent(updateReadyEvent);
+
+  t.ok(reloadRun, 'reloadRun should be true once event has been triggered');
+  t.end();
+});
+
+test('Reset the globals and remove model from localStorage scratch', t => {
```
Comment (nelsonic) [inline test/exampleComplete.spec.js:293]: looks good. as discussed. great work gents! 👍

**Review by nelsonic — APPROVED** [review]
@finnhodgkin & @ZooeyMiller great work! 🎉 
Tests and Code mods look good. 
In your _next_ PR, could you please add docs (README) for how to _run_ the tests. 
you can do this as part of: #20 👍

File: example-complete/public/script.js (hunk 1 above)
Comment (finnhodgkin) [reply example-complete/public/script.js:51]: It's referenced in #9 
textContent seems to be more supported in any case, so even without the JSDOM problem we should probably have been using it from the start.

**Pull request merged**